core/src/main/java/com/yahoo/imapnio/async/data/QResyncParameter.java

```diff
@@ -0,0 +1,92 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class models the QRESYNC parameter defined in https://tools.ietf.org/html/rfc7162#page-26.
+ */
+public class QResyncParameter {
+    /** Last known UIDVALIDITY. */
+    private long uidValidity;
+
+    /** Last known modification sequence. */
+    private long modSeq;
+
+    /** Optional set of known UIDs. */
+    private MessageNumberSet[] knownUids;
+
+    /** Optional parenthesized list of known sequence ranges and their corresponding UIDs. */
+    private QResyncSeqMatchData seqMatchData;
+
+    /**
+     * Constructor.
+     *
+     * @param uidValidity last known uidvalidity
+     * @param modSeq last known modification sequence
+     * @param knownUids known UIDs
+     * @param seqMatchData known message sequence set and their corresponding UID
+     */
+    public QResyncParameter(final long uidValidity, final long modSeq, @Nullable final List<MessageNumberSet> knownUids,
```

Comment:
why not array instead of list?  Array has the quality of  a fixed size so it wont change from callers while we are iterating.  The other thing is we have MessageNumberSet class that outputs to array directly, client will need to dump to a List in order to fit in this constructor.